Added docs for grpc at kgateway

NishantSinghhhhh opened the pull request:
```
# Description

**Motivation:** This PR addresses issue #298 by adding comprehensive GRPCRoute documentation to the kgateway.dev website. GRPCRoute is a GA resource in Gateway API v1.1.0+ and kgateway users need complete documentation to effectively route gRPC traffic in production environments.

**What changed:** 
- Added complete `content/docs/grpc-route.md` documentation file
- Includes conceptual introduction explaining GRPCRoute vs HTTPRoute for gRPC traffic
- Provides step-by-step hands-on tutorials with three practical scenarios:
  - Basic hostname-based routing with TLS termination
  - Advanced content-based routing with header and method matching
  - gRPC Reflection service integration for dynamic API exploration
- Complete API reference covering all GRPCRoute specification fields
- Operational guide with troubleshooting, cross-namespace routing patterns, and production best practices
- Route type comparison table and cleanup instructions

**Related issues:** Fixes #298

# Change Type

```
/kind documentation
```

# Changelog

```release-note
Add comprehensive GRPCRoute documentation with hands-on tutorials, API reference, and production best practices
```

# Additional Notes

- Documentation follows existing kgateway.dev Hugo structure and styling patterns
- All YAML examples are tested and production-ready
- Includes kgateway-specific features like TrafficPolicy integration
- Covers both development (self-signed certs, port-forward) and production (cloud LB, proper TLS) scenarios
- Cross-references existing kgateway documentation sections for AI Gateway, observability, and traffic management
```

NishantSinghhhhh commented:
@artberger sir and @npolshakova ma'am,

Could you kindly review the documentation changes I submitted in this PR? I would appreciate your feedback on whether any edits or improvements are needed.

Specifically, I would like to know if the length and level of detail of the documentation are appropriate or if they should be condensed for clarity and conciseness. Additionally, I welcome any suggestions to enhance the overall quality and readability of the content.

artberger commented:
@NishantSinghhhhh thanks for your work on this. I agree with your comment about how this guide could be condensed for clarity and conciseness. I have some global comments that could help with revising.

* Please review the doc contribution guide, especially for things such as wording and verbal tenses: https://kgateway.dev/docs/reference/contribution/
* The issue refers to the TCPRoute guide, which is more of the structure that I would envision, perhaps with a shorter About H2 section before the "Before you begin": https://kgateway.dev/docs/setup/listeners/tcp/
* The placement of the GRPC Route as the first topic in the nav does not make sense. I think it might be better placed elsewhere in the doc, perhaps with a different name `Traffic management > Destination types > Kubernetes services > gRPC services` --> this would be a larger structural change, so I can make that if you prefer.
* The steps I think can be shortened and structured similar to the TCPRoute page. We don't need to cover setting up kgateway and the HTTPS gateway, but can link out as step in the before you begin
* We typically do not re-document the Gateway API. I think it would be better to link out to their docs for the GRPCRoute resource in the step that you actually create the GRRCRoute.

NishantSinghhhhh commented:
> @NishantSinghhhhh thanks for your work on this. I agree with your comment about how this guide could be condensed for clarity and conciseness. I have some global comments that could help with revising.
> 
> * Please review the doc contribution guide, especially for things such as wording and verbal tenses: https://kgateway.dev/docs/reference/contribution/
> * The issue refers to the TCPRoute guide, which is more of the structure that I would envision, perhaps with a shorter About H2 section before the "Before you begin": https://kgateway.dev/docs/setup/listeners/tcp/
> * The placement of the GRPC Route as the first topic in the nav does not make sense. I think it might be better placed elsewhere in the doc, perhaps with a different name `Traffic management > Destination types > Kubernetes services > gRPC services` --> this would be a larger structural change, so I can make that if you prefer.
> * The steps I think can be shortened and structured similar to the TCPRoute page. We don't need to cover setting up kgateway and the HTTPS gateway, but can link out as step in the before you begin
> * We typically do not re-document the Gateway API. I think it would be better to link out to their docs for the GRPCRoute resource in the step that you actually create the GRRCRoute.

Okay sir I will add all the changes you have requested

NishantSinghhhhh commented:
@artberger sir Thank you for the detailed feedback! I have revised the gRPC services guide accordingly:

- Condensed content for clarity and conciseness
- Shortened About section to highlight key differences
- Structured similarly to the TCPRoute guide for simplicity
- Removed detailed kgateway/HTTPS setup; now links to existing guides
- Replaced Gateway API re-documentation with a link to official docs
- Added missing gRPC service deployment example
- Fixed certificate namespace issue for cross-namespace use
- changed the folder structure

The guide now focuses on gRPC routing concepts and links out for general setup. I’m ready for your review. Regarding the nav restructure, I’m happy for you to handle that as discussed.

Let me know if you want it adjusted further!

NishantSinghhhhh commented:
@artberger sir Thank you for the detailed feedback! I have revised the gRPC services guide accordingly:

- Condensed content for clarity and conciseness
- Shortened About section to highlight key differences
- Structured similarly to the TCPRoute guide for simplicity
- Removed detailed kgateway/HTTPS setup; now links to existing guides
- Replaced Gateway API re-documentation with a link to official docs
- Added missing gRPC service deployment example
- Fixed certificate namespace issue for cross-namespace use
- changed the folder structure

The guide now focuses on gRPC routing concepts and links out for general setup. I’m ready for your review. Regarding the nav restructure, I’m happy for you to handle that as discussed.

Can you review this PR and tell me what more changes should I add to make this PR mergeable

artberger commented:
@NishantSinghhhhh it might be better just to use a different image. Here is a different gRPC echo server and curl client we have used for other tests:

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: grpc-echo
spec:
  selector:
    matchLabels:
      app: grpc-echo
  replicas: 1
  template:
    metadata:
      labels:
        app: grpc-echo
    spec:
      containers:
        - name: grpc-echo
          image: ghcr.io/projectcontour/yages:v0.1.0
          ports:
            - containerPort: 9000
              protocol: TCP
          env:
            - name: POD_NAME
              valueFrom:
                fieldRef:
                  fieldPath: metadata.name
            - name: NAMESPACE
              valueFrom:
                fieldRef:
                  fieldPath: metadata.namespace
            - name: GRPC_ECHO_SERVER
              value: "true"
            - name: SERVICE_NAME
              value: grpc-echo
---
apiVersion: v1
kind: Service
metadata:
  name: grpc-echo-svc
spec:
  type: ClusterIP
  ports:
    - port: 3000
      protocol: TCP
      targetPort: 9000
      appProtocol: kubernetes.io/h2c
  selector:
    app: grpc-echo
---
apiVersion: v1
kind: Pod
metadata:
  name: grpcurl-client
spec:
  containers:
    - name: grpcurl
      image: docker.io/fullstorydev/grpcurl:v1.8.7-alpine
      command:
        - sleep
        - "infinity"
```
